File: lista3/zadanie2/src/Circle.cpp
#include <cmath>

#include "Figure.hpp"
#include "ClientInput.hpp"
#include "Circle.hpp"

Circle::Circle(ClientInput input) : radius(input.NextInput())
{}

double Circle::Area() 
{
    return M_PI * radius * radius;
}

double Circle::Perimeter()
{
    return M_PI * 2. * radius;
}

std::string Circle::NameYourself()
{
    return "Koło";
}

File: lista3/zadanie1/Circle.java
public class Circle extends Figure
{
    public Circle(ClientInput input)
    {
        radius = input.NextInput();
    }

    public double Area()
    {
        return Math.PI * radius * radius;
    }

    public double Perimeter()
    {
        return Math.PI * 2. * radius;
    }

    public String NameYourself()
    {
        return "Koło";
    }

    private double radius;
}
